src/frontend/Ast_to_Mir.ml

```diff
@@ -770,11 +763,15 @@ let%expect_test "read data" =
      (For (loopvar sym1__) (lower (Lit Int 1)) (upper (Lit Int 5))
       (body
        (Block
-        ((Assignment (mat ((Single (Var sym1__))))
-          (Indexed
-           (FunApp CompilerInternal FnReadData__
-            ((Lit Str mat) (Lit Str matrix) (Lit Int 10) (Lit Int 20)))
-           ((Single (Var sym1__)))))))))) |}]
+        ((For (loopvar sym2__) (lower (Lit Int 1))
+          (upper (FunApp StanLib Times__ ((Lit Int 10) (Lit Int 20))))
```

Comment:
I see here that you are indexing an Eigen matrix with a single index living in [1, n_rows * n_columns]. Does that work as a way to get the elements out?

Reply:
I'm not sure, indexing matrices at that level is broken right now though in general :D

Reply:
I mean with Eigen you can generally write `mat(i)` where `0 < i < (mat.rows() * mat.cols())`. But the code gen is broken.